sdks/java/extensions/euphoria/euphoria-core/src/test/java/org/apache/beam/sdk/extensions/euphoria/core/translate/BeamFlowTest.java

```diff
@@ -191,4 +190,232 @@ public void testPipelineWithEventTime() {
+
+    Dataset<KV<Integer, String>> output =
+        MapElements.of(inputUsedTwice).using(e -> KV.of(0, e.toString())).output();
+
+    Join.named("usage-of-input-again")
+        .of(output, inputUsedTwice)
+        .by(KV::getKey, e -> e)
+        .using((KV<Integer, String> a, Integer b, Collector<Integer> c) -> c.collect(a.getKey()))
+        .outputValues()
+        .persist(new VoidSink<>());
+
+    BeamRunnerWrapper executor = BeamRunnerWrapper.ofDirect();
+    executor.executeSync(flow);
+  }
+
+  @Test
+  public void testDatasetAsInputToMultipleTransforms() {
+    final Flow flow = Flow.create();
+    //final BeamFlow flow = BeamFlow.of(pipeline);
+
+    final ListDataSource<Integer> input =
+        ListDataSource.unbounded(Arrays.asList(1, 2, 3), Arrays.asList(2, 3, 4));
+
+    final ListDataSink<Integer> output1 = ListDataSink.get();
+    Dataset<Integer> inputUsedTwice = flow.createInput(input);
+
+    MapElements.of(inputUsedTwice).using(i -> i + 1).output().persist(output1);
+
+    Dataset<KV<Integer, String>> output =
+        MapElements.named("firstMap")
+            .of(inputUsedTwice)
+            .using(e -> KV.of(0, e.toString()))
+            .output();
+
+    Dataset<Integer> secondMapping =
+        MapElements.named("secondMap").of(inputUsedTwice).using(i -> i).output();
+
+    BeamRunnerWrapper executor = BeamRunnerWrapper.ofDirect();
+    executor.executeSync(flow);
+
+    //    PAssert.that(flow.unwrapped(output))
+    //        .containsInAnyOrder()
+
+  }
+
+  @Test
+  public void testDatasetAsInputToMultipleCoderSettingTransforms() {
+    final Flow flow = Flow.create();
+    //final BeamFlow   flow = BeamFlow.of(pipeline);
+
+    final ListDataSource<Integer> input =
+        ListDataSource.unbounded(Arrays.asList(1, 2, 3), Arrays.asList(2, 3, 4));
+
+    final ListDataSink<Integer> output1 = ListDataSink.get();
+    Dataset<Integer> inputUsedTwice = flow.createInput(input);
+
+    MapElements.of(inputUsedTwice).using(i -> i + 1).output().persist(output1);
+
+    Dataset<KV<Integer, Integer>> firstOutput =
+        ReduceByKey.named("FirstRBK")
+            .of(inputUsedTwice)
+            .keyBy(i -> i % 2)
+            .reduceBy(Sums.ofInts())
+            .windowBy(FixedWindows.of(Duration.standardSeconds(1)))
+            .triggeredBy(DefaultTrigger.of())
+            .discardingFiredPanes()
+            .output();
+
+    Dataset<KV<Integer, Integer>> secondOutput =
+        ReduceByKey.named("SecondRBK")
+            .of(firstOutput)
+            .keyBy(KV::getKey)
+            .valueBy(KV::getValue)
+            .reduceBy(Sums.ofInts())
+            .output();
+
+    BeamRunnerWrapper executor = BeamRunnerWrapper.ofDirect();
+    executor.executeSync(flow);
+
+    //    PAssert.that(flow.unwrapped(output))
+    //        .containsInAnyOrder()
+
+  }
+
+  @Test
+  @Ignore("This shout be deleted in BEAM-5248")
+  public void windowingOnceSetTwiceUsed() {
```

Comment:
Is this test of `KryoCoder`? (shouldnt be this in different class) because there is no Euphoria used so I dont fully understand the point of this test.